aiida_quantumespresso/cli/calculations/q2r.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+"""Command line scripts to launch a `Q2rCalculation` for testing and demonstration purposes."""
+from __future__ import absolute_import
+import click
+
+from aiida import orm
+from aiida.cmdline.params import options, types
+from aiida.cmdline.utils import decorators
+
+from aiida_quantumespresso.cli.utils import options as options_qe
+
+
+@click.command()
+@options.CODE(required=True, type=types.CodeParamType(entry_point='quantumespresso.q2r'))
+@options.CALCULATION(required=True)
+@options_qe.MAX_NUM_MACHINES()
+@options_qe.MAX_WALLCLOCK_SECONDS()
+@options_qe.WITH_MPI()
+@options_qe.DAEMON()
+@decorators.with_dbenv()
+def cli(code, calculation, max_num_machines, max_wallclock_seconds, with_mpi, daemon):
+    """Run a Q2rCalculation."""
+    from aiida.engine import launch
+    from aiida.plugins import CalculationFactory
+    from aiida_quantumespresso.utils.resources import get_default_options
+
+    Q2rCalculation = CalculationFactory('quantumespresso.q2r')  # pylint: disable=invalid-name
+
+    # Check that the parent calculation node comes from quantumespresso.ph.
+    # I cannot move this check into the option declaration, because CalcJobNode is not subclassed by the specific
+    # calculation plugins (only Process is), and there is no feature yet to filter by the associated process_type.
+    expected_process_type = 'aiida.calculations:quantumespresso.ph'
+    if calculation.process_type != expected_process_type:
+        raise click.BadParameter('The input calculation node has a process_type: {}; should be {}'.format(
+            calculation.process_type, expected_process_type))
+
+    parent_folder = calculation.get_outgoing(node_class=orm.RemoteData, link_label_filter='remote_folder').one().node
```

Comment:
Might be a good think to check that the calculation finished and finished ok before asking for the results.

Reply:
Good point, I think we can address this in #279 as a generic problem of validating input that is required for a restart. Although here your previous argument would also be important, that if we require the input be passed through a `calculation`, this would forbid the calculation to be launched from an input that was created otherwise, but valid nonetheless. Maybe then the option for the CLI would be better served with just directly accepting the data node itself.